src/main/java/de/bonndan/nivio/model/ComponentDiff.java

```diff
@@ -0,0 +1,61 @@
+package de.bonndan.nivio.model;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+
+public class ComponentDiff {
+
+    /**
+     * Compares changes in two strings.
+     *
+     * @param a       first
+     * @param b       second
+     * @param key     label
+     * @param changes array of changes
+     */
+    public static void compareStrings(String a, String b, String key, List<String> changes) {
+        if (!org.apache.commons.lang3.StringUtils.equals(a, b)) {
+            changes.add(key + " changed to: " + b);
+        }
+    }
+
+    /**
+     * Compares changes in string representation of two objects
+     *
+     * @param a       first
+     * @param b       second
+     * @param key     label
+     * @param changes array of changes
+     */
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    public static void compareOptionals(Optional<Object> a, Optional<Object> b, String key, List<String> changes) {
+        if (a.isEmpty() && b.isEmpty()) {
+            return;
+        }
+        if (a.isPresent() && b.isEmpty() || a.isEmpty() && b.isPresent()) {
+            changes.add(key + " changed to: " + b.orElse(""));
+            return;
+        }
+        if (!org.apache.commons.lang3.StringUtils.equals(a.map(Object::toString).orElse(""), b.map(Object::toString).orElse(""))) {
```

Comment:
Here we use the Apache Commons `StringUtils`, below we use the Spring `StringUtils` for `.isEmpty()` (line 57). Why don't we stick to just Apache Commons?

Reply:
I prefer Spring's stuff, but here I was lookingfor a null safe equality check. It turns out however that this is not necessary. Refactored.